### DISCUSSION TITLE
Where is the MA integration device?

### DISCUSSION THREAD

**Discussion by thomasdilts** [music-assistant/support#2828] [Q&A] [answered]
Lots of instructions on internet talk about an "MA integration" yet I can't find it. Even MAs own website has it.  

https://music-assistant.io/#music-assistant-integration

I've tried endlessly to find the integration but it just doesn't show up when I hit the "Add Integration" button. Obviously I have the HACS MA installed and working rather well with spotify and radio stations. I want to write some scripts but I need the MA integration I believe???
I've tried the release version, beta version and dev version of MA

**Comment by marcelveldt** [comment, marked as answer]
Music Assistant has two components:

1) The Music Assistant server (the main and must-have component) - You install this as add-on in Home Assistant, its even available in the default sources so no need to do something extra. For docker folks, we also offer a docker image.

2) While Music Assistant works perfectly fine on its own as a standalone music server, the real fun is to tie it together with Home Assistant and indeed start automating music. For that you indeed need to install the integration. At some point this will also become a standard part of Home Assistant but for now you need to install it through HACS.

**Reply by thomasdilts** [reply]
Thanks Marcel. Your reply encouraged me to at least try again to find the "Music Assistant Integration." **Finally I found it!** You see, I'm not totally new to HA. I have had HA running everything in my house for about 8 years on 5 different computers. I have helped friends install HA in their homes. My problem was everyones installation instructions for MA. They all say:

"Within HACS, search for Music Assistant and click the entry in the search results."

But I kept going to HACS--ADD ONS.  It's of course not there. It's in HACS--INTEGRATIONS. I hope this helps other people find it.

**Reply by thomasdilts** [reply]
@OzGav --- The real answer is this:

It's not in HACS--ADD ONS.  It's in HACS--INTEGRATIONS!!!

**Comment by thomasdilts** [comment]
My bad. I just saw that my HACS could be updated. Now that I have the new HACS, there is no HACS--ADD ONS or HACS--INTEGRATIONS menu. It's all just the list of integrations. Now I understand why all the instructions are like they are.

**Comment by Slim1399** [comment]
@marcelveldt Is there no way to add it to the configuration.yaml, i talking about the server installed on Docker ?

**Reply by OzGav** [reply]
Your question is not clear. Add what to configuration.yaml?

**Reply by Slim1399** [reply]
Sorry, i mean the interface of MA !

**Reply by Slim1399** [reply]
What I understood is that for the moment there is no way to have the integration of MA installed on Docker ! This is it ?

**Reply by OzGav** [reply]
The integration is a HA component. It runs on HA so it can’t run standalone in docker.

**Reply by OzGav** [reply]
If perhaps, you are using the wrong terminology, the MA server can run in docker.

**Reply by Slim1399** [reply]
First of all, thank you for your answer, bear with me :) , It's finished, I installed MA as an add-on module on a HAOS in an RPi 4 2G, it finds that 2G is extremely little for the jobe, so I was forced to switch to the other alternative "Docker", MA is installed and well connected to HA. in fact two questions that arise 
1) To avoid going back and forth between HA and MA how to integrate the interface into HA
2) Shouldn't there be a Music Assistant "Integrations", or is there no impact ?

**Reply by OzGav** [reply]
https://music-assistant.io/faq/how-to/#get-the-ma-icon-in-the-ha-sidebar

**Reply by OzGav** [reply]
And yes there is an integration which is what this original post is all about

**Reply by Slim1399** [reply]
If I understood correctly, there will be no way to do automation in this case

**Reply by OzGav** [reply]
The integration allows that

**Reply by Slim1399** [reply]
And it's for soon or it's not planned ?

**Reply by OzGav** [reply]
If you are talking about the integration that already exists
